pkg/v1/tkg/test/tkgctl/tkgs/tkgs_workload_cluster_test.go

```diff
@@ -15,162 +15,106 @@ import (
-						clusterOptions.GenerateOnly = true
-					})
-					It("should give TKC configuration as output", func() {
-						createLegacyClusterInDryRunModeTest(tkgctlClient, false, e2eConfig.WorkloadClusterOptions.ClusterName, e2eConfig.WorkloadClusterOptions.Namespace)
-					})
+					// set dry-run mode
+					clusterOptions.GenerateOnly = true
+
+					stdoutOld = os.Stdout
+					r, w, _ = os.Pipe()
+					os.Stdout = w
 				})
+				It("should give TKC configuration as output", func() {
+					Expect(err).ToNot(HaveOccurred())
 
+					w.Close()
+					os.Stdout = stdoutOld
+					var buf bytes.Buffer
+					io.Copy(&buf, r)
+					r.Close()
+					str := buf.String()
+					Expect(str).To(ContainSubstring(TKC_KIND))
+					Expect(str).To(ContainSubstring("name: " + clusterOptions.ClusterName))
+				})
 			})
 		})
 		Context("when cluster Plan is prod", func() {
 			BeforeEach(func() {
-				e2eConfig.WorkloadClusterOptions.ClusterPlan = "prod"
-				e2eConfig.WorkloadClusterOptions.ClusterName = "tkc-e2e-" + util.RandomString(4)
-				deleteClusterOptions = getDeleteClustersOptions(e2eConfig)
-				clusterOptions.ClusterConfigFile = createClusterConfigFile(e2eConfig)
-			})
-			AfterEach(func() {
-				defer os.Remove(clusterOptions.ClusterConfigFile)
+				clusterOptions.Plan = "prod"
+				clusterOptions.GenerateOnly = false
 			})
-			When("cluster class cli feature flag (features.global.package-based-lcm-beta) is set to true", func() {
-				BeforeEach(func() {
-					//set the cli feature flag as true -  (features.global.package-based-lcm-beta)
-					Expect(framework.SetCliConfigFlag(CLI_CLUSTERCLASS_FLAG, "true")).To(Succeed(), "error while setting CLI ClusterClass flag")
-					tkgctlClient, err = tkgctl.New(tkgctlOptions)
-					Expect(err).To(BeNil())
-				})
-				It("should create TKC workload cluster and delete it", func() {
-					createLegacyClusterTest(tkgctlClient, deleteClusterOptions, true, e2eConfig.WorkloadClusterOptions.ClusterName, e2eConfig.WorkloadClusterOptions.Namespace)
+
+			When("create cluster is invoked", func() {
+				AfterEach(func() {
+					err = tkgctlClient.DeleteCluster(deleteClusterOptions)
 				})
-			})
-			When("cluster class cli feature flag (features.global.package-based-lcm-beta) is set to false", func() {
-				BeforeEach(func() {
-					//set the cli feature flag as false -  (features.global.package-based-lcm-beta)
-					Expect(framework.SetCliConfigFlag(CLI_CLUSTERCLASS_FLAG, "false")).To(Succeed(), "error while setting CLI ClusterClass flag")
-					tkgctlClient, err = tkgctl.New(tkgctlOptions)
+				It("should create TKC Workload Cluster and delete it", func() {
 					Expect(err).To(BeNil())
 				})
-				It("should create TKC workload cluster and delete it", func() {
-					createLegacyClusterTest(tkgctlClient, deleteClusterOptions, false, e2eConfig.WorkloadClusterOptions.ClusterName, e2eConfig.WorkloadClusterOptions.Namespace)
-				})
 			})
 		})
 	})
+
 	Context("when input file is cluster class based", func() {
```

Comment:
I feel we should remove this clusterclass based tests from here and just keep it in the `tkgs_cc` tests directory.

Reply:
+1

Reply:
i left this here because i thought we had a negative test case. Namely if package-based-lcm-beta was false then this should error. If that is not the case, then I will just remove this test case.

Reply:
done